memory/persona.py

```
@@ -83,6 +194,237 @@ def _is_mentioned(fact_text: str, response_text: str) -> bool:
+    for pattern, explicit in _NEGATIVE_PATTERNS:
+        match = pattern.search(raw)
+        if match:
+            topic = _clean_topic_candidate(match.group("topic"))
+            if _is_specific_topic(topic):
+                return topic, explicit
+
+    explicit = any(token in raw_lower for token in _EXPLICIT_AVOID_TOKENS)
+    referenced = _extract_topic_from_reference_text(referenced_topic)
+    if explicit and referenced:
+        return referenced, True
+    # 回退：尝试从句子中剥离负面词，保留剩余较具体的主题描述。
```

Ruff (0.15.9)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)